packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -906,6 +906,64 @@ class DropdownButton<T> extends StatefulWidget {
        assert(isExpanded != null),
        assert(autofocus != null),
        assert(itemHeight == null || itemHeight >=  kMinInteractiveDimension),
+       _inputDecoration = null,
+       _isEmpty = null,
+       _isFocused = null,
+       super(key: key);
+
+  DropdownButton._({
+    Key? key,
+    required this.items,
+    this.selectedItemBuilder,
+    this.value,
+    this.hint,
+    this.disabledHint,
+    required this.onChanged,
+    this.onTap,
+    this.elevation = 8,
+    this.style,
+    this.underline,
+    this.icon,
+    this.iconDisabledColor,
+    this.iconEnabledColor,
+    this.iconSize = 24.0,
+    this.isDense = false,
+    this.isExpanded = false,
+    this.itemHeight = kMinInteractiveDimension,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.dropdownColor,
+    this.menuMaxHeight,
+    this.enableFeedback,
+    this.alignment = AlignmentDirectional.centerStart,
+    this.borderRadius,
+    InputDecoration? inputDecoration,
+    bool? isEmpty,
+    bool? isFocused,
```

Comment:
Nit.
These can be non-nullable, making the assertions redundant.

Reply:
Thanks!

Reply:
Actually, these have to be nullable so the public constructor doesn't have to override them as all final parameters have to be overridden and I don't wanna make variable mutable since it has other side affects

Reply:
I suppose I explained incorrectly. I meant doing something like below, as to begin nullable by type and non-nullable by assertion seems inconsistent. These won't be overridden in the public constructor and its private counterparts will remain unchanged. wdyt?

```suggestion
    required InputDecoration inputDecoration,
    required bool isEmpty,
    required bool isFocused,
```

Reply:
Actually, I mostly agree with @werainkhatri: the bools here should be required, and the public constructor can just explicitly set them to false (they'll be ignored anyhow if `inputDecoration` is null).

You would still have `inputDecoration` be nullable, since that indicates whether or not to apply the input decorator.

Having a nullable boolean usually means a tristate value, and these are not used as tristate values.

Reply:
Thanks, made these as required, assign false in pubic constructor